tests/callbacks/test_oom_observer.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+from packaging import version
+from torch.utils.data import DataLoader
+
+from composer import Trainer
+from composer.callbacks import OOMObserver
+from composer.trainer import Trainer
+from tests.common import RandomClassificationDataset, SimpleModel, device
+
+
+@device('cpu', 'gpu')
+def test_oom_observer_warnings_on_cpu_models(device: str):
+    if version.parse(torch.__version__) <= version.parse('2.1.0.dev'):
+        # oom_observer is supported after PyTorch 2.1.0.
+        return
+    # Error if the user sets device=cpu even when cuda is available
+    del device  # unused. always using cpu
+    with pytest.warns(UserWarning, match='The oom observer only works on CUDA devices'):
+        Trainer(
+            model=SimpleModel(),
+            callbacks=OOMObserver(),
+            device='cpu',
+            train_dataloader=DataLoader(RandomClassificationDataset()),
+            max_duration='1ba',
+        )
+
+
+@pytest.mark.gpu
+def test_oom_observer():
+    if version.parse(torch.__version__) <= version.parse('2.1.0.dev'):
+        # memory snapshot is supported after PyTorch 2.1.0.
+        return
```

Comment:
can we use pytest.skip

Reply:
unresolving since I don't think it was changed...?

Reply:
actually the correct version is to put:
```
@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.1.0'),
                    reason='OOM Observer requires PyTorch 2.1 or higher')
```
on top of function as a mark

Reply:
forgot to push. : ) . just changed it as suggested